src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -887,23 +885,24 @@ public virtual Task<bool> TryUpdateModelAsync<TModel>([NotNull] TModel model)
             [NotNull] params Expression<Func<TModel, object>>[] includeExpressions)
            where TModel : class
         {
-            if (BindingContextProvider == null)
+            if (ActionContext.BindingContext == null)
```

Comment:
This is just in case you do anything wonky. We don't expect users to see this in practice unless they are far off the beaten path.